source/common/stats/isolated_store_impl.h

```diff
@@ -77,6 +90,7 @@ template <class Base> class IsolatedStatsCache {
   StatNameHashMap<RefcountPtr<Base>> stats_;
   Allocator alloc_;
   AllocatorImportMode alloc_import_;
+  AllocatorUnit alloc_unit_;
```

Comment:
I think with this we've given up all pretense of sharing functors. Maybe just rename them all:

CounterAllocator
GaugeAllocator
HistogramAllocator

Reply:
Fixed.